src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(value) {
+    if (!value.hidden && value instanceof Herbivore) {
+      value.health -= 50;
+    }
+
+    if (value.health <= 0) {
+      Animal.alive = Animal.alive.filter((item) => item.health > 0);
```

Comment:
The logic for removing dead animals from the `Animal.alive` array could be improved. Currently, it filters the entire array every time a `Carnivore` bites a `Herbivore`. Consider removing only the specific `value` instance if its health is less than or equal to zero, which would be more efficient.